### PR TITLE
fix: gossip http-connect

### PR DESCRIPTION
**Pull request by alanshaw**
https://github.com/drand/drand/pull/496 removes the "relay from a http source" feature. This re-adds it and also adds the ability for a gRPC source to failover to http.

### CODE REVIEW

File: cmd/relay-gossip/main.go [hunk 1]
```diff
@@ -62,6 +62,16 @@ var peerWithFlag = &cli.StringSliceFlag{
 	Usage: "list of peers to connect with",
 }
 
+var httpFailoverFlag = &cli.StringSliceFlag{
```
Comment (willscott) [inline cmd/relay-gossip/main.go:65]: can this get added into the `cmd/client/lib` as part of common client construction?

File: cmd/relay-gossip/main.go [hunk 2]
```diff
@@ -105,17 +118,67 @@ var runCmd = &cli.Command{
 			defer metricsListener.Close()
 			metrics.PrivateMetrics.Register(grpc_prometheus.DefaultClientMetrics)
 		}
-		grpclient, err := grpc.New(cctx.String("grpc-connect"), cctx.String("cert"), cctx.Bool("insecure"))
+
+		var chainHash []byte
+		var err error
+
+		if cctx.String("chain-hash") != "" {
```
Comment (nikkolasg) [inline cmd/relay-gossip/main.go:125]: Can't we get the name of the flag via the variable ?

File: cmd/relay-gossip/main.go [hunk 3]
```diff
@@ -105,17 +118,67 @@ var runCmd = &cli.Command{
...
+			chainHash, err = hex.DecodeString(cctx.String("chain-hash"))
+			if err != nil {
+				return xerrors.Errorf("decoding chain hash: %w", err)
+			}
+		}
+
+		var c client.Client
+		httpConnect := cctx.StringSlice("http-connect")
```
Comment (nikkolasg) [inline cmd/relay-gossip/main.go:133]: Can't we get the name of the flag via  variable ?

File: cmd/relay-gossip/main.go [hunk 4]
```diff
@@ -105,17 +118,67 @@ var runCmd = &cli.Command{
...
+		if len(httpConnect) > 0 {
+			opts := []client.Option{}
+			if chainHash != nil {
+				opts = append(opts, client.WithChainHash(chainHash), client.WithHTTPEndpoints(httpConnect))
+			} else {
+				opts = append(opts, client.WithInsecureHTTPEndpoints(httpConnect))
+			}
+			c, err = client.New(opts...)
+		} else {
+			grpcClient, err := grpc.New(cctx.String("grpc-connect"), cctx.String("cert"), cctx.Bool("insecure"))
```
Comment (nikkolasg) [inline cmd/relay-gossip/main.go:144]: Same here ?

**Comment by alanshaw** [comment]
closing in favour of https://github.com/drand/drand/pull/563

**Pull request closed without merging**